### PR TITLE
Terraform scripts to create setup for workflow

### PR DESCRIPTION
**Pull request by parauliya**

### CODE REVIEW

**Comment by nathangoulding** [comment]
Can you rebase onto the master branch and re-open this against master?

**Comment by gauravgahlot** [comment]
@parauliya I see a few commits with the same commit message. Can we please clean this PR?

File: docker-compose.yml [hunk 1]
```diff
@@ -6,29 +6,32 @@ services:
...
+      db:
+        condition: service_healthy
     healthcheck:
       test: ["CMD-SHELL", "wget -qO- 127.0.0.1:42114/cert"]
       interval: 5s
       timeout: 2s
       retries: 30
-    volumes: 
+    volumes:
       - ./certs:/certs/${FACILITY}
```
Comment (gauravgahlot) [inline docker-compose.yml]: I believe Tinkerbell should be independent of `facility`. Though it would require change across the repo and can be a separate PR.
@nathangoulding @mmlb What do you think?

File: docker-compose.yml [hunk 2]
```diff
@@ -56,79 +61,77 @@ services:
...
       REGISTRY_HTTP_TLS_CERTIFICATE: /certs/server.pem
       REGISTRY_HTTP_TLS_KEY: /certs/server-key.pem
       REGISTRY_AUTH: htpasswd
       REGISTRY_AUTH_HTPASSWD_REALM: "Registry Realm"
       REGISTRY_AUTH_HTPASSWD_PATH: /auth/htpasswd
-    volumes: 
+    volumes:
       - ./certs:/certs
+    depends_on:
+      - fluentbit
```
Comment (gauravgahlot) [inline docker-compose.yml]: Please be consistent with either `"fluentbit"` or `fluentbit`.

File: docker-compose.yml [hunk 3]
```diff
@@ -56,79 +61,77 @@ services:
...
+      SYSLOG_BIND: $HOST_IP:514
+      TFTP_BIND: $HOST_IP:69
+      DOCKER_REGISTRY: $HOST_IP
+      REGISTRY_USERNAME: $TINKERBELL_REGISTRY_USER
+      REGISTRY_PASSWORD: $TINKERBELL_REGISTRY_PASS
+      TINKERBELL_GRPC_AUTHORITY: $HOST_IP:42113
+      TINKERBELL_CERT_URL: http://$HOST_IP:42114/cert
+      ELASTIC_SEARCH_URL: $HOST_IP:9200
     depends_on:
       - tinkerbell
```
Comment (gauravgahlot) [inline docker-compose.yml]: I don't think if `boots` should depend on `tinkerbell`. It may be applicable vice-versa though.

File: docker-compose.yml [hunk 4]
```diff
@@ -154,7 +157,45 @@ services:
       - 24224:24224
       - 24224:24224/udp
     depends_on:
-      - elasticsearch
+      - kibana
```
Comment (gauravgahlot) [inline docker-compose.yml]: Elasticsearch should not depend on Kibana.

File: docker-compose.yml [hunk 5]
```diff
@@ -154,7 +157,45 @@ services:
...
     volumes:
       - ./fluent-bit.conf:/fluent-bit/etc/fluent-bit.conf
 
+  cserver:
+    image: quay.io/packet/cacher:workflow
```
Comment (gauravgahlot) [inline docker-compose.yml]: The setup should not be using Cacher.

File: docker-compose.yml [hunk 6]
```diff
@@ -154,7 +157,45 @@ services:
...
+    environment:
+      ROLLBAR_TOKEN: ${ROLLBAR_TOKEN-9b78d0ad01d1467aa92c49c3a349b79d}
+      ROLLBAR_DISABLE: 1
+      PACKET_ENV: testing
+      PACKET_VERSION: 5efab5ef3a42cb88f2d54f4ed3201c2dd6797b7d
+      GRPC_PORT: 42115
+      HEGEL_FACILITY: ${FACILITY:-lab1}
+      HEGEL_USE_TLS: 0
+      CACHER_CERT_URL: http://127.0.0.1:42112/cert
+      CACHER_GRPC_AUTHORITY: 127.0.0.1:42111
```
Comment (gauravgahlot) [inline docker-compose.yml]: Remove Cacher environment variables

File: docker-compose.yml [hunk 7]
```diff
@@ -56,79 +61,77 @@ services:
...
-      REGISTRY_USERNAME: username
-      REGISTRY_PASSWORD: password
-      TINKERBELL_GRPC_AUTHORITY: 127.0.0.1:42113
-      TINKERBELL_CERT_URL: http://127.0.0.1:42114/cert
-      ELASTIC_SEARCH_URL: 127.0.0.1:9200
+      ROLLBAR_TOKEN: ${ROLLBAR_TOKEN:-#################################}
+      ROLLBAR_DISABLE: ${ROLLBAR_DISABLE:-1}
+      MIRROR_HOST: ${NGINX_IP:-127.0.0.1}
+      CACHER_GRPC_AUTHORITY: 127.0.0.1:42111
+      CACHER_CERT_URL: http://127.0.0.1:42112/cert
```
Comment (gauravgahlot) [inline docker-compose.yml]: Remove Cacher environment variables

File: terraform/main.tf [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  }
+  provisioner "remote-exec" {
+    script = "setup.sh"
+  }
+}
+
+# Create a device and add it to tf_project_1
+resource "packet_device" "tf-worker" {
+  hostname         = "tf-worker"
+  plan             = "c2.medium.x86"
```
Comment (gauravgahlot) [inline terraform/main.tf]: You might wanna change this. Please check once.

File: docker-compose.yml (hunk 5 above)
Comment (parauliya) [reply docker-compose.yml]: Yes, I know that and I will remove this once we get the confirmation about cacher from packet. on whether we should replace cacher with tinkerbell/hardware package.

File: docker-compose.yml (hunk 6 above)
Comment (parauliya) [reply docker-compose.yml]: Yes, I know that and I will remove this once we get the confirmation about cacher from packet. on whether we should replace cacher with tinkerbell/hardware package. Right now this is for hegel since hegel uses cacher for metadata management.

File: docker-compose.yml (hunk 7 above)
Comment (parauliya) [reply docker-compose.yml]: Same as above

File: terraform/main.tf (hunk 8 above)
Comment (parauliya) [reply terraform/main.tf]: Right now this is the only machine type I guess which support layer 2 network conversion. I am planning to put it in the input.tf file and let the user decide the machine type once we have support of layer 2 for other machine types as well.

File: docker-compose.yml (hunk 4 above)
Comment (parauliya) [reply docker-compose.yml]: I did this intentionally so that I should not call elastic, kibana and fluentbit seperately.

File: docker-compose.yml (hunk 3 above)
Comment (parauliya) [reply docker-compose.yml]: It was already there I just changed the name from server to tinkerbell.

File: docker-compose.yml (hunk 2 above)
Comment (parauliya) [reply docker-compose.yml]: I will do with the next commit.

**Comment by parauliya** [comment]
> @parauliya I see a few commits with the same commit message.
> Can we please clean this PR? May be squash a few.

I will create a new PR for this in a new branch with a single commit once we finalize about the cacher in today's call.

File: docker-compose.yml (hunk 4 above)
Comment (gauravgahlot) [reply docker-compose.yml]: Please remove it and add an additional call for Kibana.
